### ISSUE TITLE
Binary encoding is broken

### ISSUE THREAD

**Issue by ethernomad** [vweevers/lexicographic-integer-encoding#2]
The maximum value that can be encoded with "binary" is 196.

"hex" does not have this issue.

**Comment by vweevers** [comment]
How is it broken?

```js
var hex = require('lexicographic-integer-encoding')('hex')
var binary = require('lexicographic-integer-encoding')('binary')

console.log(hex.encode(197) === binary.encode(197).toString('hex'))
console.log(binary.decode(Buffer.from('c5', 'hex')))
```

```
true
197
```

**Comment by ethernomad** [comment]
This is my code:

```ts
import levelup from 'levelup'
import leveldown from 'leveldown'
import sublevel from 'subleveldown'
import lexint from 'lexicographic-integer-encoding'
import { Mutex, MutexInterface } from 'async-mutex'
let mutex: MutexInterface = new Mutex()

db = levelup(leveldown('db'))
dbEviction = sublevel(db, 'eviction', {
  keyEncoding: lexint('hex', {strict: true}),
  valueEncoding: 'ascii',
})

function storeEvictionIpfsHash(ipfsHash: string) {
  return new Promise(async (resolve, reject) => {
    let release = await mutex.acquire()
    let id: number = 0
    dbEviction.createKeyStream({
      reverse: true,
      limit: 1,
    })
    .on('data', (key: number) => {
      id = key + 1
    })
    .on('end', async () => {
      await dbEviction.put(id, ipfsHash)
      release()
      resolve()
    })
  })
}
```

With binary encoding id never gets above 196.

**Comment by vweevers** [comment]
I don't think the bug is here. Might be https://github.com/Level/subleveldown/issues/87.

Can you try it without `subleveldown`?

**Comment by vweevers** [comment]
Confirmed, `subleveldown` doesn't iterate keys that start with a byte higher than 196. Please follow aforementioned issue for updates.

**Comment by ethernomad** [comment]
Thanks for this @vweevers !

**Comment by vweevers** [comment]
Fixed in [`subleveldown@5.0.0`](https://github.com/Level/subleveldown/releases/tag/v5.0.0).
